File: v7_interface_segregation/java/Game1.java
package v7_interface_segregation.java;

public class Game1 {

  public static void main(String[] args) {
    Monster m = new Monster();
    m.basicAttack();
    MonsterBoss b = new MonsterBoss();
    b.magicAttack();
  }

}

interface BadCharacterSkill {
  public void basicAttack();
  public void magicAttack();
  public void recover();
}

class Monster implements BadCharacterSkill {
  public void basicAttack() {
    System.out.println("Monster basic attack");
  }
  public void magicAttack() {};
  public void recover() {};
}

class MonsterBoss implements BadCharacterSkill {
  public void basicAttack() {
    System.out.println("Boss basic attack");
  }
  public void magicAttack() {
    System.out.println("Boss magic attack");
  }
  public void recover() {
    System.out.println("Boss recover");
  };
}

File: v7_interface_segregation/python/game1.py
class BadCharacterSkill:
    def basicAttack(self): pass
    def magicAttack(self): pass
    def recover(self): pass

class Monster(BadCharacterSkill):
    def basicAttack(self): 
        print('monster basic attack')

class MonsterBoss(BadCharacterSkill):
    def basicAttack(self): 
        print('monsterboss basic attack')

    def magicAttack(self): 
        print('monsterboss magic attack')

    def recover(self): 
        print('monsterboss recover')

m = Monster()
m.basicAttack()

mb = MonsterBoss()
mb.magicAttack()
